packages/extension-vscode/src/quickFixProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { CodeAction, CodeActionKind, CodeActionParams, Command, TextDocuments } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { getProblemNameFromDiagnostic } from './utils/problems';
+
+export class QuickFixActionProvider {
+    private documents: TextDocuments<TextDocument>;
+    private sourceName: string;
+
+    private quickFixActionsTemplates = new Map([
+        ['vscode-webhint/ignore-hint-project', (category: string) => {
+            return `Ignore hint '${category}' in this project`;
+        }],
+        ['vscode-webhint/ignore-hint-global', (category: string) => {
+            return `Ignore hint '${category}' in all projects`;
+        }],
+        ['vscode-webhint/ignore-problem-project', (name: string) => {
+            return `Ignore problem '${name}' in this project`;
+        }],
+        ['vscode-webhint/ignore-problem-global', (name: string) => {
+            return `Ignore problem '${name}' in all projects`;
+        }]
+    ]);
+
+    public constructor(documents: TextDocuments<TextDocument>, sourceName: string) {
+        this.documents = documents;
+        this.sourceName = sourceName;
+    }
+
+    public provideCodeActions(params: CodeActionParams): CodeAction[] | null {
+        const textDocument = this.documents.get(params.textDocument.uri);
+
+        if (!textDocument) {
+            return null;
+        }
+
+        const results: CodeAction[] = [];
+
+        params.context.diagnostics.forEach((currentDiagnostic) => {
+            this.quickFixActionsTemplates.forEach((value, key) => {
+
+                // only respond to Microsoft Edge Tools requests.
+                if (!currentDiagnostic.source || currentDiagnostic.source !== this.sourceName) {
+                    return;
+                }
+
+                // default title
+                let title = currentDiagnostic.code?.toString() || '';
+
+                // if it is a problem (not a hint) use the problem as a title instead
+                if (key.includes('-problem-')) {
```

Comment:
These will need to be scoped to only the `compat-api` hints (`compat-api/css` and `compat-api/html`). Currently it matches other hint messages so things like `css-prefix-order` end up with an ignore entry, but the generated `.hintrc` doesn't work (as that hint doesn't accept an ignore list).

Reply:
This also leads to some interesting duplicate entries for our a11y hints:
![image](https://user-images.githubusercontent.com/785009/168396946-1e55d7c3-df19-46d1-8fa8-11535fca546a.png)